src/Microsoft.SqlTools.ServiceLayer/BatchParser/Lexer.cs

```diff
@@ -0,0 +1,729 @@
+﻿//------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation.  All rights reserved.
+//------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Globalization;
+using System.IO;
+
+namespace Microsoft.SqlTools.ServiceLayer.BatchParser
+{
+    internal sealed class Lexer : IDisposable
```

Comment:
>Lexer [](start = 26, length = 5)

we should add better comments to this code if we're going to port into this project...minimally we should have class level comments and method level comments for the most important public methods..ideally we'll have XML comment blocks for public methods, though that maybe overkill when porting so much code.